### PR TITLE
Bp 486 create entity mapping view

### PR DESCRIPTION
**Pull request by tommartensen**

### CODE REVIEW

File: src/CreateEntityMappingView/CreateEntityMappingView.js [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    handleEntityTypeChange(selectedEntityTypeId) {
+        this.setState({
+            selectedEntityTypeId: selectedEntityTypeId,
+            mappings: CreateEntityMappingView.getDefaultMappings()
+        });
+        this.props.lazyEntityTypeAttributeLoading(selectedEntityTypeId);
+    }
+
+    handleAddNewMappingCondition() {
+        let mappings = this.state.mappings;
```
Comment (hpibptlab) [inline src/CreateEntityMappingView/CreateEntityMappingView.js]: MINOR Make "mappings" "const". [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS3353)

File: src/CreateEntityMappingView/CreateEntityMappingView.js [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+                onChange={(event, index, value) => {this.handleEventTypeAttributeChange(key, value);}}>
+                {CreateEntityMappingView.getMenuItems(eventTypeAttributes)}
+            </SelectField>);
+        }
+        return <LoadingAnimation/>;
+    }
+
+    prepareMappingsForSending() {
+        //must be called before saving the data
+        let mappings = this.state.mappings;
```
Comment (hpibptlab) [inline src/CreateEntityMappingView/CreateEntityMappingView.js]: MINOR Make "mappings" "const". [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS3353)

File: src/CreateEntityMappingView/CreateEntityMappingView.js [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+        return <LoadingAnimation/>;
+    }
+
+    loadAttributes() {
+        const attributesFetchingIncomplete = super.render(
+            PromiseState.all(this.props.entityTypeAttributes, this.props.eventTypeAttributes));
+        if (attributesFetchingIncomplete) {
+            return attributesFetchingIncomplete;
+        }
+        let content = [];
```
Comment (hpibptlab) [inline src/CreateEntityMappingView/CreateEntityMappingView.js]: MINOR Make "content" "const". [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS3353)

File: src/CreateEntityMappingView/CreateEntityMappingView.js [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+                {CreateEntityMappingView.getMenuItems(eventTypeAttributes)}
+            </SelectField>);
+        }
+        return <LoadingAnimation/>;
+    }
+
+    prepareMappingsForSending() {
+        //must be called before saving the data
+        let mappings = this.state.mappings;
+        let cleanedMappings = [];
```
Comment (hpibptlab) [inline src/CreateEntityMappingView/CreateEntityMappingView.js]: MINOR Make "cleanedMappings" "const". [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS3353)

File: src/CreateEntityMappingView/CreateEntityMappingView.js [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+    }
+
+    handleAddNewMappingCondition() {
+        let mappings = this.state.mappings;
+        mappings[this.state.mappings.length] = {entityTypeAttribute: null, eventTypeAttribute: null};
+        this.setState({mappings});
+    }
+
+    handleEventTypeAttributeChange(key, eventTypeAttribute) {
+        let mappings = this.state.mappings;
```
Comment (hpibptlab) [inline src/CreateEntityMappingView/CreateEntityMappingView.js]: MINOR Make "mappings" "const". [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS3353)

File: src/CreateEntityMappingView/CreateEntityMappingView.js [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+                            />
+                        </div>
+                    </Container>
+                </div>
+            </div>
+        );
+    }
+}
+
+export default connect.defaults({fetch: ConnectionComponent.switchFetch})(props => ({
```
Comment (hpibptlab) [inline src/CreateEntityMappingView/CreateEntityMappingView.js]: MAJOR Remove the unused function parameter "props". [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

File: src/CreateEntityMappingView/CreateEntityMappingView.js [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+    }
+
+    handleEventTypeAttributeChange(key, eventTypeAttribute) {
+        let mappings = this.state.mappings;
+        mappings[key].eventTypeAttribute = eventTypeAttribute;
+        this.setState({mappings});
+    }
+
+    handleEntityTypeAttributeChange(key, entityTypeAttribute) {
+        let mappings = this.state.mappings;
```
Comment (hpibptlab) [inline src/CreateEntityMappingView/CreateEntityMappingView.js]: MINOR Make "mappings" "const". [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS3353)

**Comment by hpibptlab** [comment]
SonarQube analysis reported 7 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 6 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by hpibptlab** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: src/CreateEntityMappingView/CreateEntityMappingView.js [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+				this.handleEventTypeAttributeChange(key, value);
+			}}>
+			{CreateEntityMappingView.getMenuItems(eventTypeAttributes)}
+		</SelectField>);
+	}
+
+	prepareMappingsForSending() {
+		//must be called before saving the data
+		const mappings = this.state.mappings;
+		const cleanedMappings = [];
```
Comment (hpibptlab) [inline src/CreateEntityMappingView/CreateEntityMappingView.js]: MINOR Make "mappings" "const". [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS3353)

**Comment by hpibptlab** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by tommartensen** [comment]
add this to your config (is included in another open pull request)
statuses:[
        {name: "RUNNING"},
        {name: "FAILURE"},
        {name: "UNDEFINED"}
    ],

File: src/CreateEntityMappingView/CreateEntityMappingView.js [hunk 9]
```diff
@@ -0,0 +1,319 @@
...
+            if (!mapping.entityTypeAttribute.value) {
+                mapping.entityTypeAttribute.errorMessage = config.messages.requiredFieldMessage;
+				isValid = false;
+			}
+			mappings.push(mapping);
+		});
+        this.setState({
+            mappings: mappings
+        });
+		console.log(isValid);
```
Comment (hpibptlab) [inline src/CreateEntityMappingView/CreateEntityMappingView.js]: MAJOR Remove this logging statement. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS2228)

File: src/CreateEntityMappingView/CreateEntityMappingView.js [hunk 10]
```diff
@@ -0,0 +1,319 @@
...
+			});
+			isValid = false;
+        }
+        if (!this.state.selectedEventTypeId.value) {
+            this.setState({
+                selectedEventTypeId: {value: null, errorMessage: config.messages.requiredFieldMessage}
+            });
+            isValid = false;
+		}
+		let mappings = [];
```
Comment (hpibptlab) [inline src/CreateEntityMappingView/CreateEntityMappingView.js]: MINOR Make "mappings" "const". [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS3353)

**Comment by hpibptlab** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Review by julianweise — APPROVED** [review]

**Pull request merged**